ql/src/java/org/apache/hadoop/hive/ql/io/AcidUtils.java

```diff
@@ -3109,10 +3109,16 @@ public static TxnType getTxnType(Configuration conf, ASTNode tree) {
     if (tree.getFirstChildWithType(HiveParser.TOK_ALTERTABLE_COMPACT) != null){
       return TxnType.COMPACTION;
     }
+    // check if soft delete
+    if (tree.getToken().getType() == HiveParser.TOK_DROPTABLE 
+      && (HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_CREATE_TABLE_USE_SUFFIX)
+        || HiveConf.getBoolVar(conf, ConfVars.HIVE_ACID_LOCKLESS_READS_ENABLED))){
+      return TxnType.SOFT_DELETE;
```

Comment:
Shall we throw an exception if we try to drop a table but the location is not PATH_WITH_VERSION_ID?

Reply:
I don't think so as we still should be able to drop tables created without that option (not versioned)